### PR TITLE
Add mirror herb

### PR DESCRIPTION
**Pull request by ghoulslash**
mirror_herb

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Lgtm, just needs the tests :)

**Comment by ghoulslash** [comment]
Merged with upcoming and added two (passing) tests. ready for review

File: test/hold_effect_mirror_herb.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#include "global.h"
+#include "test_battle.h"
+
+ASSUMPTIONS
+{
+    gItems[ITEM_MIRROR_HERB].holdEffect == HOLD_EFFECT_MIRROR_HERB;
```
Comment (CallmeEchoo) [inline test/hold_effect_mirror_herb.c:6]:
```suggestion
    ASSUME(gItems[ITEM_MIRROR_HERB].holdEffect == HOLD_EFFECT_MIRROR_HERB);
```

File: test/hold_effect_mirror_herb.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+SINGLE_BATTLE_TEST("Mirror Herb copies all of foe's stat changes in a turn", s16 damage)
+{
+    u32 item;
+    PARAMETRIZE{ item = ITEM_NONE; }
+    PARAMETRIZE{ item = ITEM_MIRROR_HERB; }
+    GIVEN {
+        ASSUME(gItems[ITEM_MIRROR_HERB].holdEffect == HOLD_EFFECT_MIRROR_HERB);
```
Comment (CallmeEchoo) [inline test/hold_effect_mirror_herb.c:15]:
```suggestion
```
no longer needed if we make an `ASSUMPTION`

File: test/hold_effect_mirror_herb.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        EXPECT_MUL_EQ(results[0].damage, Q_4_12(1.5), results[1].damage);
+        EXPECT_EQ(player->statStages[STAT_ATK], opponent->statStages[STAT_ATK]);
+        EXPECT_EQ(player->statStages[STAT_SPEED], opponent->statStages[STAT_SPEED]);
+    }
+}
+
+SINGLE_BATTLE_TEST("Mirror Herb copies all of of Stuff Cheeks", s16 damage)
+{
+    GIVEN {
+        ASSUME(gItems[ITEM_MIRROR_HERB].holdEffect == HOLD_EFFECT_MIRROR_HERB);
```
Comment (CallmeEchoo) [inline test/hold_effect_mirror_herb.c:41]:
```suggestion
```
same as above

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**